book/listings/actions/3/main.rs

```diff
@@ -59,33 +58,32 @@ fn build_ui(app: &Application) {
         .build();
 
     // Add action "count" to `window` taking an integer as parameter
-    let action_count = SimpleAction::new_stateful(
-        "count",
-        Some(&i32::static_variant_type()),
-        &original_state.to_variant(),
-    );
-    action_count.connect_activate(clone!(@weak label => move |action, parameter| {
-        // Get state
-        let mut state = action
-            .state()
-            .expect("Could not get state.")
-            .get::<i32>()
-            .expect("The variant needs to be of type `i32`.");
+    let action_count = ActionEntry::builder("count")
+        .parameter_type(Some(&i32::static_variant_type()))
```

Comment:
Why Some? is this takes an Option??

Reply:
Yes: https://gtk-rs.org/gtk-rs-core/stable/latest/docs/gio/struct.ActionEntryBuilder.html

If you want to make me a bit happier, then please open a PR/MR at the right place 😅